src/storage/conversion/IfNeededConverter.ts

```diff
@@ -13,7 +16,7 @@ export class IfNeededConverter extends RepresentationConverter {
   private readonly converter: RepresentationConverter;
   protected readonly logger = getLoggerFor(this);
 
-  public constructor(converter: RepresentationConverter) {
+  public constructor(converter: RepresentationConverter = EMPTY_CONVERTER) {
```

Comment:
No strong opinion about this so I'll leave it up to you, but seeing as how it is used now, would it not be nicer to just not accept a converter as input parameter here and simply have this class check if the content type of the representation matches the preferences? And if not it errors and another converter will have to handle this. Could call it `MatchingConverter` or something.

Reply:
Actually, while writing this, I did like the duality. Originally I considered a refactor to _change_ `IfNeededConverter` from a wrapper around a converter to a step in a converter chain, but then I realized I did not have to choose.

After all, depending on where you plug it in, there are multiple ways:
1. Do you want to change the behavior of a single converter in the config? Just wrap it.
2. Do you want to change the behavior of a chain of converters? Just add it.

If only one of those were available, we'd need to always introduce a third component as a wrapper.